lib/graphql/relay/mongo_relation_connection.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+module GraphQL
+  module Relay
+    # A connection implementation to expose MongoDB collection objects.
+    # It works for:
+    # - `Mongoid::Criteria`
+    class MongoRelationConnection < RelationConnection
```

Comment:
:+1: I'm happily surprised how much this is able to reuse from RelationConnection! TBH it's just _not a mess_ ... and I was always afraid it would be a mess, but it's nice!